test/msdsales/paginate/main.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/amp-labs/connectors/common"
+	msTest "github.com/amp-labs/connectors/test/msdsales"
+	"github.com/amp-labs/connectors/test/utils"
+)
+
+func main() {
+	// Handle Ctrl-C gracefully.
+	ctx, done := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
+	defer done()
+
+	// Set up slog logging.
+	utils.SetupLogging()
+
+	filePath := os.Getenv("MS_SALES_CRED_FILE")
+	if filePath == "" {
+		filePath = "./ms-sales-creds.json"
+	}
+
+	conn := msTest.GetMSDynamics365SalesConnector(ctx, filePath)
+	defer utils.Close(conn)
+
+	var customPageSize int64 = 6
+
+	res, err := conn.Read(ctx, common.ReadParams{
+		ObjectName: "contacts",
+		Fields:     []string{"fullname"},
+		NextPage:   "",
+		PageSize:   customPageSize,
+		Since:      time.Now().Add(-5 * time.Minute),
+	})
+	if err != nil {
+		utils.Fail("error reading from microsoft sales", "error", err)
+	}
+
+	fmt.Println("FirstPage contacts..")
+	utils.DumpJSON(res, os.Stdout)
+
+	if len(res.NextPage) == 0 {
+		utils.Fail("there was no second page cursor")
+	}
+
+	if res.Rows != customPageSize {
+		utils.Fail(
+			fmt.Sprintf("first page has record count mismatch, given %v, expected %v", res.Rows, customPageSize))
+	}
+
+	res, err = conn.Read(ctx, common.ReadParams{
+		NextPage: string(res.NextPage), // TODO should Read params have NextPageToken type?
```

Comment:
ReadParams.NextPage(string) `<- doesn't match type with ->` ReadResult.NextPage(NextPageToken)

Reply:
Changed.
In places where it was expected to use as string instead of direct conversion I added a method. Let me know if it is not ok.
```
type NextPageToken string

func (t NextPageToken) String() string {
	return string(t)
}
```